sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -45,47 +46,60 @@ func init() {
 }
 
 func main() {
-	os.Exit(_main())
-}
-
-func _main() int {
 	// Parse command-line arguments
 	var (
 		containerID      string
 		source           string
 		containerdSocket string
 		namespace        string
 		superpowered     bool
-		pullImageOnly    bool
 	)
+	flag.Usage = func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "usage: %s run|pull-image\n", os.Args[0])
+		flag.PrintDefaults()
+	}
+	fs := flag.NewFlagSet("", flag.ExitOnError)
+	fs.StringVar(&containerdSocket, "containerd-socket", "/run/host-containerd/containerd.sock", "specifies the path to the containerd socket")
+	fs.StringVar(&namespace, "namespace", "default", "specifies the containerd namespace")
+	fs.StringVar(&source, "source", "", "the image to be pulled")
+	fs.BoolVar(&superpowered, "superpowered", false, "specifies whether to create the container with `superpowered` privileges or not")
+	fs.StringVar(&containerID, "container-id", "", "the id of the container to manage")
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, "expected one of the following subcommands: 'run', 'pull-image'")
+		flag.Usage()
+		os.Exit(2)
+	}
+	fs.Parse(os.Args[2:])
+
+	var subcommand = os.Args[1]
```

Comment:
I think that the built-in `flag` library isn't very well-suited to this subcommand-style logic.  While it'd increase the total code size to use a different library here, I think one like that will help reduce little errors that can creep in.